### ISSUE TITLE
Feedback creating Tooltip with non-dijits 

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#1271]
Tooltip is only meaningful when attached to another dijit via connectid.  It doesn't work with HTML elements or other types of widgets (though it should work with dijit-based widgets like dojox and Jon's shapes)  Right now, tooltip works by dropping on another dijit, at which point it automatically assigns the target widget an id if it does not have one and wires up the connectid.  It's meaningless to drop the tooltip on the canvas (BODY) or some non-dijit widget.

we could devise some way in the metadata of specifying that the widgetparent must be a dijit, but tagging every dijit-based widget sounds like it would be difficult to maintain. For now, if the widget is dropped on something which is not a dijit, the tooltip will be created, but not attached to anything.

@JonFerraiolo @jhpedemonte

see #1240

**Comment by jhpedemonte** [comment]
I don't think we can easily handle this declaratively.  As you stated, we would have to have some way of tagging every Dijit-based widget.

Best to do this programmatically in a helper, where the tooltip helper can check if the attached widget is Dijit-based.

**Comment by peller** [comment]
is there a hook for helpers to make this determination?

**Comment by jhpedemonte** [comment]
No, unfortunately, we'd need to create another one, pre-create.

**Comment by jhpedemonte** [comment]
I take back [what I said earlier](https://github.com/maqetta/maqetta/issues/1271#issuecomment-3026062) -- we probably can fix this declaratively.

Currently, we have the `allowedChild`/`allowedParent` metadata properties.  We could do something similar, but for "attached" widgets (as in, Tooltip is attached to a given widget; we can probably come up with a better term).  Then adding a Tooltip would only work on the specified widget types.

**Comment by jhpedemonte** [comment]
-> M6

**Comment by peller** [comment]
-> future
